### PR TITLE
[openjdk/xxx] Conan v2 migration

### PR DESCRIPTION
**Pull request by MartinDelille**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/openjdk/all/conanfile.py [hunk 1]
```diff
@@ -15,39 +16,49 @@ class OpenJDK(ConanFile):
     settings = "os", "arch"
     no_copy_source = True
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     def configure(self):
```
Comment (prince-chrismc) [inline recipes/openjdk/all/conanfile.py:19]:
```suggestion
    def validate(self):
```

File: recipes/openjdk/all/conanfile.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
-from conans import ConanFile, tools
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.tools.files import copy, get, symlinks
+from conan.errors import ConanInvalidConfiguration
 import os
 
 required_conan_version = ">=1.33.0"
```
Comment (prince-chrismc) [inline recipes/openjdk/all/conanfile.py:6]: I dont know what version symlink is but this should be close 🙈 

```suggestion
required_conan_version = ">=1.50.0"
```

File: recipes/openjdk/all/conanfile.py [hunk 3]
```diff
@@ -15,39 +16,49 @@ class OpenJDK(ConanFile):
...
         # https://github.com/conan-io/conan-center-index/pull/4491#issuecomment-774555069
-        self.copy(pattern="*", dst="conf", src=os.path.join(_source_subfolder, "conf"))
+        copy(self, pattern="*",
+                src=os.path.join(source_folder, "conf"),
+                dst=os.path.join(self.package_folder, "conf"))
 
     def package_info(self):
-        self.output.info("Creating JAVA_HOME environment variable with : {0}".format(self.package_folder))
+        self.output.info(f"Creating JAVA_HOME environment variable with : {self.package_folder}")
         self.env_info.JAVA_HOME = self.package_folder
```
Comment (prince-chrismc) [inline recipes/openjdk/all/conanfile.py:60]: This should also be added to the build env? 🤔 https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#removed-self-env-info

It's only the PATH that is automatic with the new generators

File: recipes/openjdk/all/test_package/conanfile.py [hunk 4]
```diff
@@ -1,5 +1,6 @@
-from conans import ConanFile, tools
-from conans.errors import ConanException
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.errors import ConanException
 from io import StringIO
 
 required_conan_version = ">=1.36.0"
```
Comment (prince-chrismc) [inline recipes/openjdk/all/test_package/conanfile.py:6]:
```suggestion
required_conan_version = ">=1.47.0"
```

File: recipes/openjdk/all/test_v1_package/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+from conans import ConanFile, tools
+from conans.errors import ConanException
+from io import StringIO
+
+required_conan_version = ">=1.36.0"
```
Comment (prince-chrismc) [inline recipes/openjdk/all/test_v1_package/conanfile.py:5]:
```suggestion
```

File: recipes/openjdk/all/test_v1_package/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    def test(self):
+        if not tools.cross_building(self):
+            output = StringIO()
+            self.run("java --version", output=output, run_environment=True)
+            print(output.getvalue)
+            version_info = output.getvalue()
+            if "openjdk" in version_info:
+                pass
+            else:
+                raise ConanException("java call seems not use the openjdk bin")
```
Comment (prince-chrismc) [inline recipes/openjdk/all/test_v1_package/conanfile.py:26]: little more python?

```suggestion
            if "openjdk" not in version_info:
                raise ConanException("java call seems not use the openjdk bin")
```

**Review by prince-chrismc — COMMENTED** [review]
Just some minor comments but get work!

File: recipes/openjdk/all/conanfile.py [hunk 7]
```diff
@@ -15,39 +16,49 @@ class OpenJDK(ConanFile):
     settings = "os", "arch"
```
Comment (uilianries) [inline recipes/openjdk/all/conanfile.py:16]:
```suggestion
    settings = "os", "arch", "compiler", "build_type"
```

Please, add package_id() and remove both.

File: recipes/openjdk/all/conanfile.py (hunk 1 above)
Comment (uilianries) [reply recipes/openjdk/all/conanfile.py:19]: Unfortunately we can't use `self.info.settings` here 😩

File: recipes/openjdk/all/conanfile.py (hunk 3 above)
Comment (uilianries) [reply recipes/openjdk/all/conanfile.py:60]: Yes, totally. It should be part of buildenv_info

Comment (MartinDelille) [reply recipes/openjdk/all/conanfile.py]: Done here: 4d3028d

Is it fine ?

Comment (MartinDelille) [reply recipes/openjdk/all/conanfile.py]: Ok after looking the doc, I noticed my change where partly wrong but I fixed it here: 5862740f1

**Review by prince-chrismc — DISMISSED** [review]

**Comment by MartinDelille** [comment]
Hi @jcar87! I think I misclicked and dismissed a review request that was addressed to you.

File: recipes/openjdk/all/conanfile.py [hunk 8]
```diff
@@ -12,42 +13,53 @@ class OpenJDK(ConanFile):
...
         # java.lang.SecurityException: Can't read cryptographic policy directory: unlimited
         # https://github.com/conan-io/conan-center-index/pull/4491#issuecomment-774555069
-        self.copy(pattern="*", dst="conf", src=os.path.join(_source_subfolder, "conf"))
+        copy(self, pattern="*",
+                src=os.path.join(source_folder, "conf"),
+                dst=os.path.join(self.package_folder, "conf"))
 
     def package_info(self):
-        self.output.info("Creating JAVA_HOME environment variable with : {0}".format(self.package_folder))
-        self.env_info.JAVA_HOME = self.package_folder
```
Comment (uilianries) [inline recipes/openjdk/all/conanfile.py:50]: Keep env_info.JAVA_HOME, https://conan.io/center/openapi-generator is still not ported and will not work without it.

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: recipes/openjdk/all/conanfile.py (hunk 8 above)
Comment (MartinDelille) [reply recipes/openjdk/all/conanfile.py:50]: Done here: 41fcb72a7

**Comment by MartinDelille** [comment]
Who should I ping to make this PR approved and merged ?

File: recipes/openjdk/all/conanfile.py (hunk 8 above)
Comment (MartinDelille) [reply recipes/openjdk/all/conanfile.py:50]: I thought I wrote the [openapi-generator recipe](https://github.com/conan-io/conan-center-index/pull/13691) in a conan v2 compatible way.

**Pull request merged**